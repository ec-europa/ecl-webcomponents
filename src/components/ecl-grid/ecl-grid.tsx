import { Component, Prop, h, Element } from '@stencil/core';

@Component({
  tag: 'ecl-grid',
  styleUrls: {
    ec: './build/styles/ecl-grid-ec.css',
    eu: './build/styles/ecl-grid-eu.css',
  },
  shadow: false,
  scoped: false,
})

export class EclGrid {
  @Element() el: HTMLElement;
  @Prop() styleClass: string = '';
  @Prop() theme: string = 'ec';
  @Prop() columns: number = 12;
  @Prop() breakpoint: string;
  @Prop() row: boolean = false;
  @Prop() container: boolean = false;

  getClass(): string {
    let styleClasses = [];

    if (this.row || this.container) {
      if (this.row) {
        styleClasses = [
          `ecl-row`,
          this.styleClass,
        ];
      } else {
         styleClasses = [
          `ecl-container`,
          this.styleClass,
        ];
      }
    } else {
      if (!this.breakpoint) {
        styleClasses = [
          `ecl-col-${this.columns}`,
          this.styleClass,
        ];
      } else {
        styleClasses = [
          `ecl-col-${this.breakpoint}-${this.columns}`,
          this.styleClass,
        ];
      }
    }
    
    return styleClasses.join(' ');
  }

  componentDidRender() {
    const parent = this.el.parentNode;
    while (this.el.firstChild) parent.insertBefore(this.el.firstChild, this.el);
    parent.removeChild(this.el);
  }

  render() {
    return (
      <div class={this.getClass()}>
        <slot></slot>
      </div>
    )
  }
}
