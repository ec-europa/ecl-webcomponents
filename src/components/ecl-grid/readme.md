# ecl-grid



<!-- Auto Generated Below -->


## Properties

| Property     | Attribute     | Description | Type     | Default |
| ------------ | ------------- | ----------- | -------- | ------- |
| `columns`    | `columns`     |             | `number` | `12`    |
| `styleClass` | `style-class` |             | `string` | `''`    |
| `theme`      | `theme`       |             | `string` | `'ec'`  |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
