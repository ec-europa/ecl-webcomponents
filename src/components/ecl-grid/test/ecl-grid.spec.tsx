import { newSpecPage } from '@stencil/core/testing';
import { EclGrid } from '../ecl-grid';

describe('ecl-grid', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclGrid],
      html: `<div class="ecl-container">
              <ecl-grid columns="6">Content in 6 columns</ecl-grid>
              <ecl-grid columns="4">Content in 4 columns</ecl-grid>
              <ecl-grid columns="2">Content in 2 columns</ecl-grid>
            </div>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
