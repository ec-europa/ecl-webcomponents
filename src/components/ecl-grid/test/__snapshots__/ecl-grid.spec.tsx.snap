// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-grid renders 1`] = `
<ecl-grid class="ecl-col-6" columns="6">
  <!---->
  Content in 6 columns
</ecl-grid>
`;
